lib/storage/src/content_manager/consensus/consensus_wal.rs

```diff
@@ -74,13 +78,171 @@ impl ConsensusOpWal {
         Ok(entry.transpose()?)
     }
 
+    /// Difference between raft index and WAL record number.
+    /// Difference might be different because of consensus snapshot.
+    fn index_offset(&self) -> Result<Option<u64>, StorageError> {
```

Comment:
Can we use it also in `fn entry(&self, id: u64)`? There we also calculate the offset.

Reply:
unfortunately no, it performs some extra intermediate operations